executor/executor.go

```diff
@@ -427,18 +429,43 @@ func (executor *Executor) execute(
 
 				if runningPods == targetPodCount {
 					msg = statusMap[status]
+					result = proto.DecisionExecutionStatusSucceed
 					break
 				}
 			}
 		}
 
-		executor.logger.Infof(ctx, msg, "time: ", time.Now(), " .... ", start)
+		//rollback in case of failed to restart all pods
+		if runningPods < targetPodCount {
+
+			msg = statusMap[kv1.PodFailed]
+			result = proto.DecisionExecutionStatusFailed
+
+			memoryLimit := container.Resources.Limits.Memory().Value()
+			memoryRequest := container.Resources.Requests.Memory().Value()
+			cpuLimit := container.Resources.Limits.Cpu().MilliValue()
+			cpuRequest := container.Resources.Requests.Cpu().MilliValue()
+
+			*totalResources.Containers[0].Limits.Memory = memoryLimit / 1024 / 1024
+			*totalResources.Containers[0].Requests.Memory = memoryRequest /1024 / 1024
+			*totalResources.Containers[0].Limits.CPU = cpuLimit
+			*totalResources.Containers[0].Requests.CPU = cpuRequest
+
+			// execute the decision with old values to rollback
+			_, err := executor.kube.SetResources(kind, name, namespace, totalResources)
+
+			if err != nil {
+				executor.logger.Infof(ctx, "can't rollback decision")
```

Comment:
should be Warnf log not info

Reply:
do we need to keep going if an error occured in this step?

Reply:
yes, to return the status

Reply:
change the `Infof` to `Warnf`

Reply:
CHANGED